task_list_app/task_list_app/src/main/java/com/jackson/task_list_app/api/controller/TaskController.java

```diff
@@ -1,51 +1,90 @@
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
 import com.jackson.task_list_app.api.models.Task;
+import com.jackson.task_list_app.api.models.TaskStatus;
 import com.jackson.task_list_app.api.services.TaskService;
 
-
-
 @RestController
 public class TaskController {
 
     @Autowired
     private TaskService taskService;
 
+    // GetMappings
     @GetMapping("/tasks")
     @ResponseStatus(HttpStatus.OK)
-    public List<Task> getTask() {
-        return taskService.getTask();
+    public List<Task> getTasks(@RequestParam(required = false) String status) {
+        if (status != null) {
+            try {
+                TaskStatus taskStatus = TaskStatus.valueOf(status.toUpperCase());
+                return taskService.getTasksByStatus(taskStatus);
+            } catch (IllegalArgumentException e) {
+                // Handle invalid status value
+                return Collections.emptyList();
+            }
+        } else {
+            return taskService.getTask();
+        }
     }
 
+    // Post Mappings
     @PostMapping("/tasks")
     public ResponseEntity<Task> createTask(@RequestBody Task newTask) {
         Task createdTask = taskService.createTask(newTask);
         return new ResponseEntity<>(createdTask, HttpStatus.CREATED);
     }
 
+    // Put Mappings
     @PutMapping("/tasks/{id}")
-    public ResponseEntity<Task> updateTask(@PathVariable Long id, @RequestBody Task updatedTask) {
+    public ResponseEntity<Task> updateTask(
+            @PathVariable Long id,
+            @RequestBody Task updatedTask,
+            @RequestParam(required = false) TaskStatus newStatus) {
         Task existingTask = taskService.getTaskById(id);
 
         if (existingTask != null) {
             updatedTask.setId(id);
             Task task = taskService.updateTask(updatedTask);
+
+            if (newStatus != null) {
+                existingTask.setStatus(newStatus);
+                task = taskService.updateTask(existingTask);
+            }
+
             return new ResponseEntity<>(task, HttpStatus.CREATED);
         } else {
             return new ResponseEntity<>(HttpStatus.NOT_FOUND);
         }
     }
+
+    // Delete Mappings
+    @DeleteMapping("/tasks/{id}/delete")
+    public ResponseEntity<Task> deleteTask(@PathVariable Long id) {
+        Task existingTask = taskService.getTaskById(id);
+
+        if (existingTask != null) {
+            existingTask.setStatus(TaskStatus.DELETED);
+            Task updatedTask = taskService.updateTask(existingTask);
+            return new ResponseEntity<>(updatedTask, HttpStatus.CREATED);
```

Comment:
Please change this to a `HTTP 202 Accepted`. Your `DELETE` request isn't creating anything. You can set that `HTTP 202` status through a `@ResponseStatus()` annotation and shorten this to:

```suggestion
            return taskService.updateTask(existingTask);
```

Reply:
I changed it to ACCEPTED, but I do have a question on how to shorten it properly.

Reply:
I tried myself and it doesn't work the way I thought without a `@ExceptionHandler` this is something I will teach you (or you can teach yourself). It's a great thing to know. This looks great.